### PR TITLE
This adds --enable-re2 so we can benefit from libre2 if we want

### PR DESCRIPTION
**Pull request by ahupowerdns**
Plus documents it. RE2Rule()

### CODE REVIEW

**Comment by ahupowerdns** [comment]
@rubenk did I get the autoconf stuff right?

File: pdns/dnsdistdist/m4/pdns_check_re2.m4 [hunk 1]
```diff
@@ -0,0 +1 @@
+../../../m4/pdns_check_re2.m4
```
Comment (Habbie) [inline pdns/dnsdistdist/m4/pdns_check_re2.m4]: do you imagine needing/wanting re2 in auth later?

Comment (ahupowerdns) [inline pdns/dnsdistdist/m4/pdns_check_re2.m4]: no but this still allows me to make dnsdist from the main directory if needed

**Comment by pieterlexis** [comment]
autotools changes look good from here.

File: pdns/dnsdistdist/m4/pdns_check_re2.m4 (hunk 1 above)
Comment (Habbie) [inline pdns/dnsdistdist/m4/pdns_check_re2.m4]: that's a feature I feel should go in general, it's a maintenance pain - and now our auth tarball knows about re2!

Comment (pieterlexis) [inline pdns/dnsdistdist/m4/pdns_check_re2.m4]: I am inclined to agree with @Habbie on this (although I understand both viewpoints)

**Pull request merged**